### PR TITLE
[duktape] Replace the old way of using the port

### LINKED ISSUE #35308: [duktape] fix find_package_handle_standard_args or instructions
### Is your feature request related to a problem? Please describe.

The instructions for the duktape package contain:

```cmake
find_package(duktape CONFIG REQUIRED)
```

However, `duktapeConfig.cmake` contains the following code:

```cmake
FIND_PACKAGE_HANDLE_STANDARD_ARGS(Duktape
    REQUIRED_VARS DUKTAPE_LIBRARY DUKTAPE_INCLUDE_DIR)
```

which leads to the following warning:

Windows:

```
CMake Warning (dev) at C:/Program Files/.../cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (Duktape)
  does not match the name of the calling package (duktape).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
```

Linux:

```
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (Duktape)
  does not match the name of the calling package (duktape).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
```

If I replace the find_package command with:

```cmake
find_package(Duktape REQUIRED)
```

then things seem to work locally for some reason but only Windows works in CI. This is what I get on Linux:

```
CMake Error at /opt/hostedtoolcache/vcpkg/master/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  By not providing "FindDuktape.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Duktape", but
  CMake did not find one.

  Could not find a package configuration file provided by "Duktape" with any
  of the following names:

    DuktapeConfig.cmake
    duktape-config.cmake

  Add the installation prefix of "Duktape" to CMAKE_PREFIX_PATH or set
  "Duktape_DIR" to a directory containing one of the above files.  If
  "Duktape" provides a separate development package or SDK, be sure it has
  been installed.
```

For this reason I'm now using

```cmake
set(DUKTAPE_PACKAGE_NAME duktape)
if (WIN32)
    set(DUKTAPE_PACKAGE_NAME Duktape)
endif()
find_package(${DUKTAPE_PACKAGE_NAME} REQUIRED)
```

which works both locally and in CI, although I have it still gives me the warning locally.

### Proposed solution

It's hard to propose a solution because I get errors or warnings with both duktape and Duktape. I believe it's just a matter of fixing the inconsistency in these scripts. The way cmake handles these capitalized names always gives me a headache and I'm not always sure what the best alternative is.

### Describe alternatives you've considered

```cmake
set(DUKTAPE_PACKAGE_NAME duktape)
if (WIN32)
    set(DUKTAPE_PACKAGE_NAME Duktape)
endif()
find_package(${DUKTAPE_PACKAGE_NAME} REQUIRED)
```

which is obviously ugly and still leads to warnings.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Cheney-W**
Fixes https://github.com/microsoft/vcpkg/issues/35308

- Replace the old usage added in 2019 with the currently preferred usage.
- `duktape` only has one downstream, which is `dukglue`, and `dukglue` is also a header-only port.

Usage passed with following triplets:
x64-windows 
x64-windows-static 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/duktape/CMakeLists.txt [hunk 1]
```diff
@@ -25,31 +24,26 @@ if (BUILD_SHARED_LIBS)
...
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file("${PROJECT_BINARY_DIR}/duktapeConfigVersion.cmake"
   COMPATIBILITY SameMajorVersion
 )
 
-configure_file(duktapeConfig.cmake.in "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/duktapeConfig.cmake" @ONLY)
-
 install(FILES
-  "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/duktapeConfig.cmake"
   "${PROJECT_BINARY_DIR}/duktapeConfigVersion.cmake"
```
Comment (LilyWangLL) [inline ports/duktape/CMakeLists.txt:47]:
```suggestion
  "${PROJECT_BINARY_DIR}/unofficial-duktapeConfigVersion.cmake"
```

File: ports/duktape/CMakeLists.txt [hunk 2]
```diff
@@ -25,31 +24,26 @@ if (BUILD_SHARED_LIBS)
...
+install(EXPORT unofficial-duktapeConfig
+  FILE unofficial-duktapeConfig.cmake
+  NAMESPACE unofficial::duktape::
+  DESTINATION "share/unofficial-duktape"
 )
 
-export(PACKAGE duktape)
-
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file("${PROJECT_BINARY_DIR}/duktapeConfigVersion.cmake"
```
Comment (LilyWangLL) [inline ports/duktape/CMakeLists.txt:42]:
```suggestion
write_basic_package_version_file("${PROJECT_BINARY_DIR}/unofficial-duktapeConfigVersion.cmake"
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Review by LilyWangLL — APPROVED** [review]

**Review by dg0yt — COMMENTED** [review]
Again, IMO there should be a grace period for customers which use the old CMake config name.

**Comment by Cheney-W** [comment]
> Again, IMO there should be a grace period for customers which use the old CMake config name.

Thank you for your comment.
Could you please elaborate on what `a grace period` means? Are you referring to providing a method for users to transition from the old usage to the new usage?

**Comment by Cheney-W** [comment]
Duplicate of https://github.com/microsoft/vcpkg/pull/35813

**Pull request closed without merging**